searx/settings.yml

```diff
@@ -3,7 +3,7 @@ general:
     instance_name : "searx" # displayed name
 
 search:
-    safe_search : 0 # Filter results. 0: None, 1: Moderate, 2: Strict
+    safe_search : 2 # Filter results. 0: None, 1: Moderate, 2: Strict
```

Comment:
Thanks for taking up this discussion.

> Most (if not all) of the search engines (including Google) don't enforce safe search by default 

That's correct for search engines that earn their money with tracking. DuckDuckGo by example does safe search by default.   Even better if you type xXx at DDG they give you only one link, direct to the google-Image search.

> Moreover, there are more adults than kids on the world

A very technical view with a fail. Do really all adults searching for vulgar content .. all the time? Looking at the real world:  Bottles with toxic contents have safety closures .. by default ;)

> Finally, if you are worried that your child see bad content then enforce safe search on your own Searx instance :).

Correct, that is exactly the point we discuss!  User can change the setting in the preferences and admins of searx instances can can set a default in the `settings.yml`.

Therefore setting the default in the master branch is just a standing and the question I want to discuss is; for what do we, the searx community  stand?

One thing we stand for (I think) is "neutrality" and therefore the default should not filter any content, on the other side I see "responsibility" and what basic attitude we have about.

> Conclusion: From my point of view, you can change back the safe_search parameter to 0.

let's see what others say.

Reply:
Searx can't guarantee that safe search really filters everything and it might have false positives which does not look great when someone tries it for the first time and does not find the results he would expect on google.

If you personally think that safe search should be on by default you can do that on your instance but changing this default setting is out of scope of this PR.

Reply:
> Searx can't guarantee that safe search really filters everything 

"False positives" are inherent of all filters. We just using the "save--browsing" filters from the origin engines.

> If you personally think that safe search should be on by default you can do that on your instance

It is not about my personal feeling, the question was:
 
> Therefore setting the default in the master branch is just a standing and the question I want to discuss is; for what do we, the searx community stand?
> One thing we stand for (I think) is "neutrality" and therefore the default should not filter any content, on the other side I see "responsibility" and what basic attitude we have about.

And I'm not the only one asking this question see "related issues" mentioned in the commit messages:

- #636
- #1907
- #1392
- #1410

Reply:
> And I'm not the only one asking this question see "related issues" mentioned in the commit messages:
> 
> - #636
> - #1907
> - #1392
> - #1410
> 

I have looked at all these issues and none of them clearly want to have safesearch **on** by default in Searx itself. Only requesting to have an adult content filter feature feature in Searx which for some engine like Google already does it for a long time.
Maybe I'm missing something here but all these issues could be closed by saying just change safesearch in the settings.yml or in the preferences page.

Moreover, I have doubt that 3 issues represent all the Searx community. I can see myself seeing issues like "how to disable safesearch" or "please disable safesearch by default" appearing after enabling safesearch by default in the **core of Searx**.

Finally, I would prefer that you open a separate PR for enabling safesearch by default instead of silently trying to modify a big change in another PR that isn't really about filtering adult content. This would also allow to go ahead and merge this PR because quite a lot of users would like to have google engine fixed, which by the way thank you for that :).

Reply:
> Finally, I would prefer that you open a separate PR for enabling safesearch by default instead of silently trying to modify a big change in another PR that isn't really about filtering adult content. 

Yes, that was very unfair of me, but I know that a separate PR is not discussed anyway ;)

I will revert the patch, but my question is right open:

> Therefore setting the default in the master branch is just a standing and the question I want to discuss is; for what do we, the searx community stand?
> One thing we stand for (I think) is "neutrality" and therefore the default should not filter any content, on the other side I see "responsibility" and what basic attitude we have about.

Reply:
Reverted commit 4e4173f - the old safe-search default is now back (and disabled by  default), see f32f1e8. I will send a PR to enable safe search by default!